deps: bump cockroachdb/errors and cockroachdb/redact

knz opened the pull request:
Fixes #53207
Fixes #53700
Needed for #53312 

This picks up improved support for `%T` in format strings,
as well as better support for `%w`.

Release justification: low risk, high benefit changes to existing functionality

Release note: None

knz committed:
deps: bump cockroachdb/errors and cockroachdb/redact

This picks up improved support for `%T` in format strings,
as well as better support for `%w`.

Release justification: low risk, high benefit changes to existing functionality

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/53764)
<!-- Reviewable:end -->

irfansharif reviewed (approved):
LGTM, though looks like we'll have to update TestRedactedDecodeFile.

knz committed:
colexecerror: require an `error` argument for InternalError()

Prior to this patch, `colexecerror.InternalError()` would take any
type as input. This was encouraging callers to pass not only constant
strings and errors, it was also encouraging uses of `fmt.Sprintf`

However, given that these errors ultimately make their way to crash
reporting, we really want as much details as possible in crash
reports. Simple strings don't cut it and are considered fully unsafe
for reporting.

What we want instead if a discrete separation of safe and unsafe
details.  The machinery to do this is available in the `redact`
package, but for ease of use the `errors` library also does it.

So this patch changes `InternalError()` to require an `error`
argument, and invokes `errors.AssertionFailedf` or `log.PanicAsError`
in the call points.

This API can be further simplified in the future by making
`InternalError`
responsible for constructing the error object, for example

```go
InternalErrorf(format string, args ...interface{}) {
  panic(errors.AssertionFailedf(format, args...))
}
```

This improvement is left for a later change though.

Release justification: low risk, high benefit changes to existing functionality

Release note: None

knz committed:
*: convert more panic strings to error objects

Error objects in panics allow us to expose more safe details for
telemetry reporting.

Release justification: low risk, high benefit changes to existing functionality

Release note: None

knz commented:
bors r=irfansharif
